src/App.tsx

```diff
@@ -1,61 +1,138 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { User } from './type/User';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+function chooseKey() {
+  const maxKey = [...todos].sort((a, b) => b.id - a.id);
+
+  return maxKey[0].id + 1;
+}
+
+const addTodo = (title: string, userId: number): void => {
+  const Todo = {
+    id: chooseKey(),
+    title,
+    userId,
+    completed: false,
+    user: getUser(userId),
+  };
+
+  todos.push(Todo);
```

Comment:
No. We should use internal component state for managing todos.
Use `useState` hook